src/bp/nlu-core/training-pipeline.ts

```diff
@@ -118,13 +120,16 @@ const PreprocessInput = async (input: TrainInput, tools: Tools): Promise<TrainSt
   } as TrainStep
 }
 
-const makeListEntityModel = async (entity: ListEntity, botId: string, languageCode: string, tools: Tools) => {
+const makeListEntityModel = async (entity: ListEntityWithCache, languageCode: string, tools: Tools) => {
   const allValues = _.uniq(Object.keys(entity.synonyms).concat(..._.values(entity.synonyms)))
   const allTokens = (await tools.tokenize_utterances(allValues, languageCode)).map(toks =>
     toks.map(convertToRealSpaces)
   )
 
-  return <ListEntityModel>{
+  const cache = new LRUCache(1000)
```

Comment:
I don't mind doing this, original implementation tried to keep training pipeline not aware of what cache implementation we use (in this case LRUCache), pretty much the whole purpose of the cacheManager.

If you don't have any concerns having the training pipeline depend on a specific cache implementation, then you're good

Reply:
I don't agree that whole purpose of the cache manager was to hide of an LRUCache behind an abstraction... The previous cache-manager always used `LRUCache` for interface even though it was hidding the "new" call.

The cache manager was also the container of actual floating caches:

`const cacheMap: _.Dictionary<LRUCache<string, any>> = {}`

I got rid of it, because I wanted the cache to be passed to the training-pipeline explicitely, not just retrieved from a static function.

But you're right that it is weird. Maybe I should extract the call in another function? something like :

cache-warmer.ts
```
export interface ICache { // So training pipeline is not aware of actual cache implementation.
    get(key: string)
    set(key: string, value: any)
}

export function warm-cache(cacheEntries: LRUCache.Entry[]): ICache {
    const cache = new Cache(1000)
    cache.load(cacheEntries)
    return entries
}
```

What do you think?